posts/inside-rust/0000-00-00-const-eval-roadmap-and-skill-tree.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: "Constant evaluation roadmap"
+author: Oli Scherer
+team: the const eval working group <https://github.com/rust-lang/const-eval/>
+---
+
+Y'all probably noticed that const eval finally got loops, mutable local variables and branches. With those features existing, we were able to make a lot of basic arithmetic functions `const fn `. Things like `checked_div` needs to be able to check that the divisor is not zero and return `None` in that case. These features have been asked for a lot, and while it may not appear so, these were the low hanging fruit.
```

Comment:
I don't understand the purpose of the 2nd sentence here ("Things like..."). It seems unrelated to the first and third sentence, and just cutting it doesn't leave a gap as far as I can see.

Reply:
I didn't know where to put an example... Maybe I could just say

> basic arithmetic functions (e.g. `checked_div`) `const fn`

?

Reply:
Yes, that seems much better. Also `checked_div` seems rather niche, isn't there something "bigger"?

Reply:
I think `unwrap_or` works now and didn't before, anything with Option or Result really.

Reply:
I chose `checked_div` because it's pretty obvious what it does. the `Option`/`Result` methods may still have some blockers somewhere, not sure if we fixed/stabilized the drop fixes needed for that.

Reply:
What about "(e.g. checked and saturating arithmetic)"? That makes it less specific to "div".
(I am looking at https://blog.rust-lang.org/2020/08/27/Rust-1.46.0.html to see what got stabilized there.)